packages/legend-graph/src/graphManager/AbstractPureGraphManager.ts

```diff
@@ -490,6 +490,7 @@ export abstract class AbstractPureGraphManager {
   abstract getQueryInfo(queryId: string): Promise<QueryInfo>;
   abstract createQuery(query: Query, graph: PureModel): Promise<Query>;
   abstract updateQuery(query: Query, graph: PureModel): Promise<Query>;
+  abstract renameQuery(queryId: string, queryName: string): Promise<void>;
```

Comment:
why not use `updateQuery`?

Reply:
`updateQuery` would take graph as parameter. If we are renaming a query we won't have the graph info of that query so i thought i would add a new function

Reply:
honestly, `updateQuery` and `createQuery` don't need the graph.

Reply:
let's unify this to use `updateQuery` for now

Reply:
We do a validation after we update the query. We try to check if we can find the mapping and runtime mentioned in the updated query in the current graph. Do u want me to remove that validation then?

Reply:
there is another issue. We have only `LightQuery` here. If we want to use `updateQuery` then we need to have `Query` info. Either ways i have to introduce a new function to handle this